### PR TITLE
Use ProblemDiagnostics for Plugin Id and remove TaskPathLocationTransformer

### PR DESCRIPTION
**Pull request by reinsch82**
Fixes
 - #27949 

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test ACT

File: platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy [hunk 1]
```diff
@@ -254,12 +256,7 @@ class JavaCompileProblemsIntegrationTest extends AbstractIntegrationSpec {
         assert details: "Expected details to be non-null, but was null"
 
         def locations = problem["locations"] as List<Map<String, Object>>
-        assert locations.size() == 2: "Expected two locations, but received ${locations.size()}"
-
-        def taskLocation = locations.find {
-            it.containsKey("buildTreePath")
-        }
-        assert taskLocation != null: "Expected a task location, but it was null"
+        assert locations.size() == 1: "Expected two locations, but received ${locations.size()}"
```
[the remark this answers is not in the document]
Comment (reinsch82) [reply platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy:259]: that is a good catch 🎉

**Comment by reinsch82** [comment]
@bot-gradle test this

File: platforms/ide/problems/src/main/java/org/gradle/problems/internal/transformers/ProblemStreamLocationTransformer.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.gradle.problems.ProblemDiagnostics;
+import org.gradle.problems.buildtree.ProblemStream;
+
+public class ProblemStreamLocationTransformer implements ProblemTransformer {
+
+    private final ProblemStream problemStream;
+
+    public ProblemStreamLocationTransformer(
+        ProblemStream problemStream
+    ) {
```
Comment (hegyibalint) [inline platforms/ide/problems/src/main/java/org/gradle/problems/internal/transformers/ProblemStreamLocationTransformer.java:34]:
```suggestion
    public ProblemStreamLocationTransformer(ProblemStream problemStream) {
```
💄

**Review by hegyibalint — APPROVED** [review]
One cosmetic comment overall, nice work!
I hope my space-age assertion method didn't cause too much pain.

**Comment by reinsch82** [comment]
@bot-gradle merge

**Comment by cobexer** [comment]
@bot-gradle test RFN

**Comment by reinsch82** [comment]
@bot-gradle merge

**Pull request merged**